webapp/src/index.js

```diff
@@ -16,12 +17,24 @@ class Plugin {
     // eslint-disable-next-line no-unused-vars
     initialize(registry, store) {
         registry.registerChannelHeaderButtonAction(
```

Comment:
Why not make this dependent on the server version?

Reply:
Good suggestion

Reply:
@larkox @hanzei Note that focalboard is doing something similar instead of bumping the server version. It is checking whether `registry.registerProduct` exists before calling it. https://github.com/mattermost/focalboard/blob/e518c391aa26496abe253ac7048a33a3f254da10/mattermost-plugin/webapp/src/index.tsx#L134

We would be bumping all plugins to `6.3` if we go the route of bumping the server version

Reply:
We should also have community plugins in mind here. [Standup Raven](https://github.com/standup-raven/standup-raven) currently has its min server version set to `5.25.0`. Bumping to `6.3.0` is a big jump for something that can potentially be backwards compatible

Reply:
I am not talking about bumping the server version, but conditionally showing things depending on the server version. Something like:
```
if (minimunSeverVersion(6.3)) {
   registerAppBar()
} else {
   registerChannelHeader()
}
```

Reply:
@hanzei @larkox Handling the conditional registering like this makes sense, but I'm thinking this makes testing the feature more difficult, as we would need to have a 6.3 release branch to be cut before being able to test the individual plugins right? I'm not sure what are the full implications of relying on server version in this regard. I agree that it certainly makes the code more readable. Do you have any advice on how to make it testable while using the server version conditional?

Also note that we will still be registering the channel header button in the plugin. The webapp now has logic to filter out channel header buttons meeting this criteria: If a given plugin registers components in both locations, and the app bar is enabled, that plugin's channel header buttons will be hidden.

This is done this way as the app bar can be toggled through feature flag, and the webapp needs to know about the channel header components so it can still show those when appropriate. This makes it so only one piece of code (the webapp itself, not the individual plugins) is in charge of deciding what to show in the UI.